### DISCUSSION TITLE
如何预设代理和API

### DISCUSSION THREAD

**Discussion by DearLicy** [asadahimeka/pixiv-viewer#10] [问题反馈]

**Comment by asadahimeka** [comment]
可通过设置环境变量 `VUE_APP_DEF_HIBIAPI_MAIN` 和 `VUE_APP_DEF_PXIMG_MAIN`
格式参照
![image](https://github.com/asadahimeka/pixiv-viewer/assets/31837214/4f3b7e44-7023-45d6-ac40-9e4a43189dc1)
![image](https://github.com/asadahimeka/pixiv-viewer/assets/31837214/9beed7fb-81bb-4dd1-b229-974e8b547df4)

**Comment by yabo083** [comment]
@asadahimeka 
Hello author, I would like to ask how to configure it inside the program besides this manual setting on the web page? I have created the .env.development file in the root directory of the project, and in it I have configured: 
``` 
NODE_ENV=development 
VUE_APP_PXIMG_PROXYS='i.pixiv.re' 
VUE_APP_HIBIAPI_ALTS='https://api.obfs.dev/api/pixiv' 
``` 

This shows up with the close manual proxy button, but the page still doesn't load properly, so I'd like to ask you what your solution is. How can I configure the proxy address in the code?

**Comment by asadahimeka** [comment]
@yabo083, if you have created an `.env` file, you need to set it like this:
```
VUE_APP_DEF_PXIMG_MAIN=i.pixiv.re
VUE_APP_DEF_HIBIAPI_MAIN =https://api.obfs.dev/api/pixiv
```
Also, if you wanna configure these links directly in the code, you may search `process.env.VUE_APP_DEF_HIBIAPI_MAIN` & `process.env.VUE_APP_DEF_PXIMG_MAIN` globally and replace it with your owns in your editor.
The key replacement code is located in https://github.com/asadahimeka/pixiv-viewer/blob/master/src/api/http.js#L5 and https://github.com/asadahimeka/pixiv-viewer/blob/master/src/api/index.js#L20.

And there is a CORS issue about `https://api.obfs.dev/api/pixiv` if you wanna use it in this site, you need to solve it yourself.

**Comment by yabo083** [comment]
ok! thank you very much for your reply, this program is great for beginners and I will look into it closely.

**Comment by asadahimeka** [comment]
> ok! thank you very much for your reply, this program is great for beginners and I will look into it closely.

Thank you for your interest.
